internal/operators/cnv/cnv_operator.go

```diff
@@ -129,18 +138,51 @@ func (o *operator) GetMonitoredOperator() *models.MonitoredOperator {
 }
 
 // GetHostRequirements provides operator's requirements towards the host
-func (o *operator) GetHostRequirements(_ context.Context, _ *common.Cluster, host *models.Host) (*models.ClusterHostRequirementsDetails, error) {
+func (o *operator) GetHostRequirements(ctx context.Context, _ *common.Cluster, host *models.Host) (*models.ClusterHostRequirementsDetails, error) {
+	log := logutil.FromContext(ctx, o.log)
 	switch host.Role {
 	case models.HostRoleMaster:
 		return &models.ClusterHostRequirementsDetails{
-			CPUCores: masterCPU,
-			RAMMib:   masterMemory,
+			CPUCores: MasterCPU,
+			RAMMib:   MasterMemory,
 		}, nil
 	case models.HostRoleWorker, models.HostRoleAutoAssign:
+		overhead, err := o.getDevicesMemoryOverhead(host)
+		if err != nil {
+			log.WithError(err).WithField("inventory", host.Inventory).Errorf("Cannot parse inventory for host %v", host.ID)
+			return nil, err
+		}
 		return &models.ClusterHostRequirementsDetails{
-			CPUCores: workerCPU,
-			RAMMib:   workerMemory,
+			CPUCores: WorkerCPU,
+			RAMMib:   WorkerMemory + overhead,
 		}, nil
 	}
 	return nil, fmt.Errorf("unsupported role: %s", host.Role)
 }
+
+func (o *operator) getDevicesMemoryOverhead(host *models.Host) (int64, error) {
+	if host.Inventory == "" {
+		return 0, nil
+	}
+	inventory, err := hostutil.UnmarshalInventory(host.Inventory)
+	if err != nil {
+		return 0, err
+	}
+	gpuCount := o.getGPUCount(*inventory)
+	// One GPU imposes 1GiB of additional memory requirement
+	return conversions.GibToMib(gpuCount), nil
+}
+
+func (o *operator) getGPUCount(inventory models.Inventory) int64 {
+	var gpuCount int64
+	for _, gpu := range inventory.Gpus {
+		if o.config.SupportedGPUs[getDeviceKey(gpu)] {
```

Comment:
In case it will be changed to a list of strings, should change here to `funk.Contains`